lib-find-meal/src/main/java/com/kinandcarta/lib/find/meal/view/FindMealFragment.kt

```diff
@@ -73,141 +42,61 @@ class FindMealFragment : Fragment() {
     }
 
-    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
-        super.onActivityResult(requestCode, resultCode, data)
-        if ((requestCode == REQUEST_CHECK_SETTINGS) && (resultCode == Activity.RESULT_OK)) {
-            locationUpdateState = true
+    private fun setupUIWithView(view: View) {
+        val mealsRecyclerView: RecyclerView = view.findViewById(R.id.rvMeals)
+        mealsRecyclerView.adapter = mealsAdapter
+        mealsRecyclerView.layoutManager = LinearLayoutManager(context)
+        swipeRefreshLayout.setOnRefreshListener {
+            swipeRefreshLayout.isRefreshing = false
+            viewModel.updateMeals(true)
         }
     }
 
-    override fun onPause() {
-        super.onPause()
-        fusedLocationClient.removeLocationUpdates(locationCallback)
-    }
-
-    override fun onResume() {
-        super.onResume()
-        if (!locationUpdateState) {
-            startLocationUpdates()
+    private fun onStateChanged(state: FindMealViewModel.State) {
+        when (state) {
+            FindMealViewModel.State.LoadingMeals ->
+                progressBarView.isVisible = true
+            is FindMealViewModel.State.LoadedMeals ->
+                onLoadedMeals(state)
+            is FindMealViewModel.State.Failed ->
+                onFailure(state.failure)
         }
     }
 
-    override fun onDestroy() {
-        super.onDestroy()
-        mainScope.cancel() // TODO: We'd get this for free in a viewModel?
+    private fun onLoadedMeals(state: FindMealViewModel.State.LoadedMeals) {
+        progressBarView.isVisible = false
+        mealsAdapter.submit(state.meals, state.distanceUnit)
     }
 
-    private fun updateMeals(needReload: Boolean) {
-        val location = lastLocation ?: return
-        progressBarView.isVisible = true
-
-        mainScope.launch {
-            kotlin.runCatching {
-                sdk.getSortedMeals(location.latitude, location.longitude, distanceUnit, needReload)
-            }.onSuccess {
-                mealsAdapter.mealsList = it
-                mealsAdapter.notifyDataSetChanged()
-            }.onFailure {
-                Toast.makeText(context, it.localizedMessage, Toast.LENGTH_SHORT)
-                    .show()
+    private fun onFailure(failure: FindMealViewModel.Failure) {
+        when (failure) {
+            is FindMealViewModel.Failure.LoadingMealsFailed -> {
+                progressBarView.isVisible = false
+                showToast(failure.localizedMessage ?: "An unexpected error occurred loading meals")
             }
-            progressBarView.isVisible = false
         }
     }
 
     companion object {
-        private val LOCATION_PERMISSION_REQUEST_CODE = 1
-        private val REQUEST_CHECK_SETTINGS = 2
-        fun newInstance() = FindMealFragment()
+        private const val LOCATION_PERMISSION_REQUEST_CODE = 1
+        private const val REQUEST_CHECK_SETTINGS = 2 // TODO: Need to re-add this check somewhere?
```

Comment:
This isn't being used because I took the code out, wasn't sure if we definitely need it still, but we can re-add if necessary. If not, I'll remove this, I just left it here to remind us to think about it

Reply:
I think with the refactoring that we can remove it now.